### PR TITLE
🎉 New Source: xkcd [python cdk]

### PR DESCRIPTION
**Pull request by sarafonseca-123**
## What
Add a new source: xkcd.

## How
Using [Python CDK Speedrun](https://docs.airbyte.com/connector-development/tutorials/cdk-speedrun/) as base.

## Recommended reading order
1. `airbyte-integrations/connectors/source-xkcd/source_xkcd/source.py`
2. `airbyte-integrations/connectors/source-xkcd/source_xkcd/spec.yaml`
3. `airbyte-integrations/connectors/source-xkcd/source_xkcd/schemas/xkcd.json`
4. `airbyte-integrations/connectors/source-xkcd/README.md`
5. `airbyte-integrations/connectors/source-xkcd/bootstrap.md`

## 🚨 User Impact 🚨
There are no breaking changes, the only impact will be a new source to gather data from.

## Pre-merge Checklist

<summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [x] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [x] Documentation updated
    - [x] Connector's `README.md`
    - [x] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [x] `docs/integrations/README.md`
    - [x] `airbyte-integrations/builds.md`
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

## Tests

<details><summary><strong>Unit</strong></summary>

image

</details>

<details><summary><strong>Acceptance</strong></summary>

</details>

### CODE REVIEW

**Comment by sarafonseca-123** [comment]
Hi! I'd like some help with this connector. The xkcd API iterates through updating the path, passing the comic number eg: `https://xkcd.com/{comic_num}/json.html`. I chose to make this connector by:
- Making the first requisition to https://xkcd.com/info.0.json, which returns the last comic available
- In the function parse_response, I extract the comic number from this response
- I proceed to get all comics in the range of 1 - last_comic_number through a custom function _get_comic_metadata.

This works, but I'm getting a timeout from the acceptance test basic_read. It times out in 300 secs, and my connector is running in 8 minutes. Is there a better way to work this connector?
I also thought of making the comic id a connection spec but would limit the connector to 1 specified comic.

**Comment by marcosmarxm** [comment]
@sarafonseca-123 please check your commits looks you made one with another account. We need all accounts to sign the CLA.

**Comment by sarafonseca-123** [comment]
@marcosmarxm fixed!

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
@sarafonseca-123 looks good but you must change the logic to use functions in the correct step.

File: airbyte-integrations/connectors/source-xkcd/source_xkcd/source.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        stream_slice: Mapping[str, Any] = None,
+        next_page_token: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping]:
+        num_records = response.json().get("num")
+        for num in range(1, num_records):
+            try:
+                record = self._get_comic_metadata(num)
+            except Exception:
+                continue
+            yield record
```
[the remark this answers is not in the document]
Comment (sarafonseca-123) [reply airbyte-integrations/connectors/source-xkcd/source_xkcd/source.py]: I implemented a new iteration method, more in line with other connectors.

File: airbyte-integrations/connectors/source-xkcd/source_xkcd/source.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+
+class XkcdStream(HttpStream):
+    url_base = "https://xkcd.com"
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+    
+    def path(self, **kwargs) -> str:
+        return "/info.0.json"
```
[the remark this answers is not in the document]
Comment (sarafonseca-123) [reply airbyte-integrations/connectors/source-xkcd/source_xkcd/source.py]: I implemented a new iteration method, more in line with other connectors.

**Comment by sarafonseca-123** [comment]
@marcosmarxm I improved the iteration strategy to be more like the [salesforce](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py) one. Sorry about the trouble!

**Comment by marcosmarxm** [comment]
@sarafonseca-123 can you share the acceptance test?

**Review by marcosmarxm — APPROVED** [review]
thanks for the contribution @sarafonseca-123 !

**Comment by RealChrisSean** [comment]
https://github.com/airbytehq/connector-contest/issues/89

**Pull request merged**